doc/whats_new/v1.4.rst

```diff
@@ -35,6 +35,11 @@ random sampling procedures.
       solvers (when fit on the same data again). The amount of change depends on the
       specified `tol`, for small values you will get more precise results.
 
+- |Efficiency| :class:`preprocessing.QuantileTransformer` now uses `resample` function to
```

Comment:
We need to check that we don't change the output of the `KBinsDiscretizer` as well. We might be calling another NumPy API.

Reply:
It indeed looks like
```py
np.random.RandomState(0).choice(n_samples, size=subsample, replace=False)
```
gives the same sampling as
```py
a = np.arange(n_samples)
np.random.RandomState(0).shuffle(a)
a[:subsample]
```

probably what's used underneath. I'm okay with this change, although here it has no impact on efficiency.